libraries/rawkit/src/lib.rs

```diff
@@ -138,13 +149,38 @@ impl RawImage {
 		};
 
 		raw_image.camera_model = Some(camera_model);
-		raw_image.transform = transform;
+		raw_image.orientation = orientation;
 
 		raw_image.calculate_conversion_matrices();
 
 		Ok(raw_image)
 	}
 
+	/// Extracts the thumbnail image from the raw file.
+	pub fn extract_thumbnail<R: Read + Seek>(reader: &mut R) -> Result<ThumbnailImage, DecoderError> {
+		let mut file = TiffRead::new(reader)?;
+		let ifd = Ifd::new_first_ifd(&mut file)?;
+
+		// TODO: ARW files Store the thumbnail offset and length in the first IFD. Add support for other file types in the future.
```

Comment:
Would it be worthwhile to do an assert on `metadata::identify::identify_camera_model`?